src/xcode/ENA/ENA/Source/Views/BaseElements/ENATanInput.swift

```diff
@@ -19,32 +19,41 @@ import Foundation
 import UIKit
 
 protocol ENATanInputDelegate: AnyObject {
-	func tanChanged(isValid: Bool)
+	func tanChanged(isValid: Bool, checksumIsValid: Bool, isBlocked: Bool)
 }
 
 @IBDesignable
 class ENATanInput: UIControl, UIKeyInput {
-	@IBInspectable var textColor: UIColor = UIColor.preferredColor(for: .textPrimary1)
-	@IBInspectable var boxColor: UIColor = UIColor.preferredColor(for: .backgroundSecondary)
+	@IBInspectable var labelTextColor: UIColor = .enaColor(for: .textPrimary1)
+	@IBInspectable var boxColor: UIColor = .enaColor(for: .separator)
 
 	@IBInspectable var fontSize: CGFloat = 30
-	@IBInspectable var digits: Int = 7
+	@IBInspectable var groups: String = "3,3,4"
+	@IBInspectable var forbidden: String = "1,0,I,O,L,Ä,Ö,Ü, ,-,/,:,;,(,),€,&,@,\""
```

Comment:
What about all other characters? The whole world of unicode is open. Should this rather be a whitelist character set?